File: Practice/src/coding_basic_training/ArrangeIntegersInSpiral.js
function solution(n) {
    var answer = [];
    for (let i = 0; i < n; i++) {
        answer.push(Array(n));
    }
    for (let v = 1, l = 0, r = n - 1, u = 0, d = n - 1; l <= r && u <= d; ) {
        for (let i = l; i <= r; i++) {
            answer[u][i] = v++;
        }
        u++;
        for (let i = u; i <= d; i++) {
            answer[i][r] = v++;
        }
        r--;
        for (let i = r; i >= l; i--) {
            answer[d][i] = v++;
        }
        d--;
        for (let i = d; i >= u; i--) {
            answer[i][l] = v++;
        }
        l++;
    }
    return answer;
}

File: Practice/src/coding_basic_training/ArrangeIntegersInSpiral.py
def solution(n):
    answer = [[0] * n for _ in range(n)]
    d = [[0, 1], [1, 0], [0, -1], [-1, 0]]
    x = 0
    y = 0
    c = 0
    for i in range(1, n * n + 1):
        answer[x][y] = i
        x2 = x + d[c % 4][0]
        y2 = y + d[c % 4][1]
        if x2 < 0 or y2 < 0 or x2 >= n or y2 >= n or answer[x2][y2] != 0:
            c += 1
        x += d[c % 4][0]
        y += d[c % 4][1]
    return answer
